---
title: "Terminal"
---

# SFTP Commands
## The Command


```bash
sftp -oPort=22 -oHostKeyAlgorithms=+ssh-rsa guest2@131.174.197.250

```

## 🧠 Why do we need -oHostKeyAlgorithms=+ssh-rsa?
💡 Problem:
Modern versions of OpenSSH (since v8.8) have disabled the ssh-rsa algorithm by default due to security concerns. So when a server only supports ssh-rsa (and not newer, safer algorithms), your SSH/SFTP client refuses to connect.
You were seeing an error like:

```code 
Unable to negotiate with 131.174.197.250 port 22: no matching host key type found. Their offer: ssh-rsa,ssh-dss
```

## 🔐 What this means:
The server only supports ssh-rsa or ssh-dss, both of which are considered old/insecure.

Your SSH client refused to accept them for security reasons.

## ✅ What does -oHostKeyAlgorithms=+ssh-rsa do?
It tells your SSH/SFTP client:

"It's okay to accept and use ssh-rsa for host verification."

This lets you bypass the modern security restriction just for this connection, allowing it to proceed.


## 🔒 Security Note:
This is generally safe for trusted servers, especially in closed environments (like HPC systems).

But don’t use this option on unknown/public servers unless necessary, because older algorithms like ssh-rsa are more vulnerable to certain attacks.

## Downloading the Folder in SFTP
```code
get -r foldername
```

# To download files from BIH
```python
rsync bihlogin1:/data/cephfs-1/home/users/alhassa_m/work/run_nf/rare/dag.png dag.png
```

# Speedey Downloads
```python
aria2c -x16 -s16 https://ftp.ensembl.org/pub/release-114/variation/indexed_vep_cache/homo_sapiens_vep_114_GRCh38.tar.gz
```


Following are the parameters needed for not skipping any workflows.
```python
 17     vep_cache                      = params.pipelines_testdata_base_path + 'vep_cache_and_plugins.tar.gz'
 18     vcfanno_toml                   = params.pipelines_testdata_base_path + 'vcfanno_config.toml'
 19     vcfanno_resources              = params.pipelines_testdata_base_path + 'vcfanno_resources.txt'
 20     gnomad_af                      = params.pipelines_testdata_base_path + 'gnomad_reformated.tab.gz'
 21     score_config_snv               = params.pipelines_testdata_base_path + 'rank_model_snv.ini'
 22     variant_consequences_snv       = params.pipelines_testdata_base_path + 'variant_consequences_v2.txt'
 23     svdb_query_dbs                 = params.pipelines_testdata_base_path + 'svdb_querydb_files.csv'
 24     score_config_sv                = params.pipelines_testdata_base_path + 'rank_model_sv.ini'
 25     variant_consequences_sv        = params.pipelines_testdata_base_path + 'variant_consequences_v2.txt'
 26     vep_filters                    = params.pipelines_testdata_base_path + 'hgnc.txt'
 27     score_config_mt                = params.pipelines_testdata_base_path + 'rank_model_snv.ini'
 28     mobile_element_svdb_annotations = params.pipelines_testdata_base_path+ 'svdb_querydb_files.csv'
 29     mobile_element_references       = params.pipelines_testdata_base_path+ 'mobile_element_references.tsv'
```


```python
# Needed by the tool germlinecnvcaller
params.ploidy_model not set.
params.gcnvcaller_model not set.
params.readcount_intervals not set.
```

```python
# Tool removed from skip list

```


```python
# Work Fine without any extra parameters
haplogrep3
qualimap

```


## Tools with Errors
### smncopynumbercaller
There was error with chr number.


### NGS-Bits
```python
Plus 73 more processes waiting for tasks…
Execution cancelled -- Finishing pending tasks before exit
-[nf-core/raredisease] Pipeline completed with errors-
ERROR ~ Error executing process > 'NFCORE_RAREDISEASE:RAREDISEASE:QC_BAM:NGSBITS_SAMPLEGENDER (hugelymodelbat)'

Caused by:
  Process `NFCORE_RAREDISEASE:RAREDISEASE:QC_BAM:NGSBITS_SAMPLEGENDER (hugelymodelbat)` terminated with an error exit status (1)


Command executed:

  SampleGender \
      -in hugelymodelbat_sorted_md.bam \
      -method xy \
      -out hugelymodelbat_ngsbits_sex.tsv \
      -ref reference.fasta \


  cat <<-END_VERSIONS > versions.yml
  "NFCORE_RAREDISEASE:RAREDISEASE:QC_BAM:NGSBITS_SAMPLEGENDER":
      ngs-bits: $(echo $(SampleGender --version 2>&1) | sed 's/SampleGender //' )
  END_VERSIONS

Command exit status:
  1

Command output:
  (empty)

Command error:
  INFO:    Converting SIF file to temporary sandbox...
  SampleGender 2023_02
  Exception: Chromosome 'chrX' not known in BAM/CRAM file hugelymodelbat_sorted_md.bam
  Location : ../../src/cppNGS/BamReader.cpp:606
  INFO:    Cleaning up image...

Work dir:
  /data/cephfs-1/work/groups/kircher/users/alhassa_m/run_nf/rare/workdir/8b/e936ad41edf167e3351710509c090e

Container:
  /data/cephfs-1/work/groups/kircher/users/alhassa_m/run_nf/rare/workdir/singularity/depot.galaxyproject.org-singularity-ngs-bits-2023_02--py311ha0b7adc_2.img

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line

 -- Check '.nextflow.log' file for details
ERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting

 -- Check '.nextflow.log' file for details
```



